packages/upload-client/src/types.ts

```diff
@@ -95,6 +125,18 @@ export interface ProgressStatus extends XHRProgressStatus {
 export type ProgressFn = (status: ProgressStatus) => void
 
 export interface Service extends StorefrontService {
+  ucan: {
+    conclude: ServiceMethod<
+      UCANConclude,
+      UCANConcludeSuccess,
+      UCANConcludeFailure
+    >
+  }
+  blob: {
+    add: ServiceMethod<BlobAdd, BlobAddSuccess, BlobAddFailure>
+    remove: ServiceMethod<BlobRemove, BlobRemoveSuccess, BlobRemoveFailure>
```

Comment:
Argh, I completely forgot, we're also going to need `blob/get` for the console. Could you send a follow up PR to add? So sorry.

Reply:
I don't really see this [capability in the spec](https://github.com/w3s-project/specs/blob/main/w3-blob.md). How should we go about this?

Reply:
Ah yes it'll need a space PR also.

Essentially the same as `store/get` - it allows the caller to get the size of the blob (and potentially other stuff in the future). The allocations storage already has a method, you just need to create a handler, call it and return it: https://github.com/w3s-project/w3up/blob/9982d12b0a1f9a6da3f0d4264b9a35348e189dfb/packages/upload-api/src/types/blob.ts#L20-L23